src/plugins/syscalls/syscalls.cpp

```diff
@@ -128,10 +130,110 @@ static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
     return 0;
 }
 
+static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j,nargs;
+    unsigned char* buf; // pointer to buffer to hold argument values
+
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls *s = wrapper->sc;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    if(wrapper->syscall_index>-1) { // get arguments only if we know how many to get
+        nargs = win_syscall_struct[wrapper->syscall_index].num_args;
+        unsigned long size = s->reg_size * nargs;
+
+        buf = (unsigned char *)g_malloc(sizeof(char)*size);
+
+        if(s->reg_size==4){ // 32 bit os
+            ctx.addr = info->regs->rsp + s->reg_size;  // jump over base pointer
+
+            // multiply num args by 4 for 32 bit systems to get the number of bytes we need
+            // to read from the stack.  assumes standard calling convention (cdecl) for the
+            // visual studio compile.
+            if((size_t)size != vmi_read(vmi, &ctx, buf, size)){
+                goto exit;
+            }
+        }
+        else { // 64 bit os - ************** UNTESTED *******************!
+	    uint64_t *buf64 = (uint64_t *)buf;
+            if(nargs > 0) {
+                buf64[0] = info->regs->rcx;
+            }
+            if(nargs > 1) {
+                buf64[1] = info->regs->rdx;
+            }
+            if(nargs > 2) {
+                buf64[2] = info->regs->r8;
+            }
+            if(nargs > 3) {
+                buf64[3] = info->regs->r9;
+            }
+            if(nargs>4) { // first 4 agrs passed via rcx, rdx, r8, and r9
+                ctx.addr = info->regs->rsp+0x20;  // jump over homing space
+                unsigned long sp_size = s->reg_size * (nargs-4);
+                if((size_t)sp_size != vmi_read(vmi, &ctx, &(buf64[4]), sp_size)){
+                    goto exit;
+                }
+           }
+        }
+    }
+
+    switch(s->format) {
+    case OUTPUT_CSV:
+        printf("syscall,%" PRIu32" 0x%" PRIx64 ",%s,%" PRIi64 ",%s,%s",
+               info->vcpu, info->regs->cr3, info->procname, info->userid, info->trap->breakpoint.module, info->trap->name);
+        if(wrapper->syscall_index>-1) { // only print arguments if we got them
+            printf(",Arguments:%d,",nargs);
+            for(i=0;i<nargs;i++) {
+                printf("%s,0x",win_syscall_struct[wrapper->syscall_index].args[i].name);
+                for(j=s->reg_size-1;j>=0;--j) { // j must be signed
+                    printf("%02X", buf[i*s->reg_size+j]);  // bytes stored in reverse order, switch them here
```

Comment:
You are just accessing the memory using the wrong pointer type here (char instead of uint32_t/uint64_t)